### PR TITLE
[ts-sdk] Add `create-live-compositor` package

### PR DESCRIPTION
**Pull request by wkozyra95**
### Public API changes

- Remove `LiveCompositor.create` function. Instead user has to
```
const compositor = new LiveCompositor();
await compositor.init()
```
Old API is technically safer, but it make hard to assign compositor globally which is a common pattern in js. (See express example)

### Other

- Add create command that can create basic node project or larger example with express and redux
- Some base implementation to provide structure to adding different types of projects

### CODE REVIEW

File: ts/create-live-compositor/src/options.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+const packageManagers: Choice<PackageManager>[] = [
+  { value: 'npm', title: 'npm' },
+  { value: 'yarn', title: 'yarn' },
+  { value: 'pnpm', title: 'pnpm' },
+];
+
+export async function resolveOptions(): Promise<ProjectOptions> {
+  const projectName = await textPrompt('Project name: ', 'live-compositor-app');
+  // TODO: replace
```
Comment (wkozyra95) [inline ts/create-live-compositor/src/options.ts:33]: Unused but I don't want to revert all the browser aware code

File: ts/create-live-compositor/templates/node-express-redux/src/store.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  initialState: true,
+  reducers: create => ({
+    toggle: create.reducer(state => !state)
+  }),
+  selectors: {
+    shouldShow: state => state,
+  }
+})
+
+// `combineSlices` automatically combines the reducers using
```
Comment (wkozyra95) [inline ts/create-live-compositor/templates/node-express-redux/src/store.ts:15]: all of the comments here are from redux template, I decided to left them, but I don't have a strong opinon on that

File: ts/create-live-compositor/src/options.ts [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    runtimeOptions = await resolveNodeOptions();
+  } else {
+    throw new Error('Unknown runtime');
+  }
+
+  return {
+    runtime: runtimeOptions,
+    packageManager,
+    projectName,
+    directory: path.join(process.cwd(), 'app'),
```
Comment (noituri) [inline ts/create-live-compositor/src/options.ts:55]: I would expect that the project is created at `./projectName` not `./app`

File: ts/create-live-compositor/templates/node-express-redux/package.json [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    "react": "^18.3.1",
+    "react-redux": "^9.1.2"
+  },
+  "devDependencies": {
+    "@types/express": "^4.17.21",
+    "@types/node": "^20.14.10",
+    "@types/react": "^18.2.25",
+    "typescript": "^5.5.3"
+  }
+}
```
Comment (noituri) [inline ts/create-live-compositor/templates/node-express-redux/package.json:26]: I'm not sure if it's on purpose or not but there is no publicly available `0.1.0-rc.0` version. The latest versions released on npm are `0.1.0-rc.1` or `0.0.1`

File: ts/create-live-compositor/templates/node-express-redux/src/App.tsx [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+function Instructions() {
+  return (
+    <View direction="column">
+      <View />
+      <Text fontSize={50}>Open index.ts and get started.</Text>
+      <View height={20} />
+      <Text width={960} fontSize={30} wrap="word">
+        This example renders static text and sends the output stream via RTP to local port
+        8001. Generated code includes a helpers in liveCompositorFfplayHelper.ts that display output
```
Comment (noituri) [inline ts/create-live-compositor/templates/node-express-redux/src/App.tsx:37]:
```suggestion
        8001. Generated code includes helpers in liveCompositorFfplayHelper.ts that display output
```

File: ts/create-live-compositor/templates/node-express-redux/src/liveCompositorFfplayHelper.ts [hunk 6]
```diff
@@ -0,0 +1,59 @@
+import os from 'os';
+import path from 'path';
+import fs from 'fs';
+import { ChildProcess, spawn as nodeSpawn, SpawnOptions } from 'child_process';
+
+const TMP_SDP_DIR = path.join(os.tmpdir(), 'live-composiotor-sdp');
+
+/**
+ * Util function that displays video send over RTP to specified port.
```
Comment (noituri) [inline ts/create-live-compositor/templates/node-express-redux/src/liveCompositorFfplayHelper.ts:9]:
```suggestion
 * Util function that displays video sent over RTP to the specified port.
```

File: ts/create-live-compositor/templates/node-minimal/src/liveCompositorFfplayHelper.ts [hunk 7]
```diff
@@ -0,0 +1,59 @@
+import os from 'os';
+import path from 'path';
+import fs from 'fs';
+import { ChildProcess, spawn as nodeSpawn, SpawnOptions } from 'child_process';
+
+const TMP_SDP_DIR = path.join(os.tmpdir(), 'live-composiotor-sdp');
+
+/*
+ * Util function that displays video send over RTP to specified port.
```
Comment (noituri) [inline ts/create-live-compositor/templates/node-minimal/src/liveCompositorFfplayHelper.ts:9]:
```suggestion
 * Util function that displays video sent over RTP to the specified port.
```

File: ts/create-live-compositor/templates/node-minimal/src/App.tsx [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+
+function App() {
+  return (
+    <View direction="column">
+      <View />
+      <Text fontSize={50}>Open index.ts and get started</Text>
+      <View height={20} />
+      <Text width={1000} fontSize={30} wrap="word">
+        This example renders static text and sends the output stream via RTP to local port
+        8001. Generated code includes a helpers in liveCompositorFfplayHelper.ts that display output
```
Comment (noituri) [inline ts/create-live-compositor/templates/node-minimal/src/App.tsx:13]:
```suggestion
        8001. Generated code includes helpers in liveCompositorFfplayHelper.ts that display output
```

**Comment by wkozyra95** [comment]
I needed to publish this to test publishing and versioning flow and after the publish it did not make sense to wait for any feedback here, so I merged it. Feel free to post a review, feedback will be included in the follow up.

File: ts/create-live-compositor/templates/node-express-redux/src/liveCompositorFfplayHelper.ts
Comment (WojciechBarczynski) [inline ts/create-live-compositor/templates/node-express-redux/src/liveCompositorFfplayHelper.ts:1]: This filename looks so wierd. I get that it follows naming convention, but it just looks wierd.

**Review by WojciechBarczynski — COMMENTED** [review]
LGTM

File: ts/create-live-compositor/templates/node-express-redux/src/liveCompositorFfplayHelper.ts
Comment (wkozyra95) [reply ts/create-live-compositor/templates/node-express-redux/src/liveCompositorFfplayHelper.ts:1]: I think this is good. This is something user should remove

**Pull request merged**